docs/pages/environment_creation/index.md

```diff
@@ -5,116 +5,329 @@ title: Environment Creation
+target on the grid that has been placed randomly at the beginning of the episode.
+
+- Observations provide the location of the target and agent. 
+- There are 4 actions in our environment, corresponding to the movements "right", "up", "left", and "down".  
+- A done signal is issued as soon as the agent has navigated to the grid cell where the target is located.
+- Rewards are binary and sparse, meaning that the immediate reward is always zero, unless the agent has reached the target, then it is 1
 
-We will first write the code for our custom environment in `gym-foo/gym_foo/envs/foo_env.py` - let's call it `FooEnv`. All custom environments should subclass `gym.Env` and override the `step`, `reset`, `render`, `close`  methods like so:
+An episode in this environment (with `size=5`) might look like this:
 
+<img src="https://user-images.githubusercontent.com/15806078/160155148-253a05ae-25c1-4fcf-9a72-f72362a64225.gif" width="35%">
+
+where the blue dot is the agent and the red square represents the target.
+
+
+Let us look at the source code of `GridWorldEnv` piece by piece: 
+
+### Declaration and Initialization
+Our custom environment will inherit from the abstract class `gym.Env`. You shouldn't forget to add the `metadata` attribute to you class. 
+There, you should specify the render-modes that are supported by your environment (e.g. `"human"`, `"rgb_array"`, `"ansi"`)
+and the framerate at which your environment should be rendered.
+In `GridWorldEnv`, we will support the modes "rgb_array" and "human" and render at 4 FPS.
+
+The `__init__` method of our environment will accept the integer `size`, that determines the size of the square grid.
+We will set up some variables for rendering and define `self.observation_space` and `self.action_space`.
+In our case, observations should provide information about the location of the agent and target on the 2-dimensional grid. 
+We will choose to represent observations in the form of a dictionaries with keys `"agent"` and `"target"`. An observation
+may look like ` {"agent": array([1, 0]), "target": array([0, 3])}`.
+Since we have 4 actions in our environment ("right", "up", "left", "down"), we will use `Discrete(4)` as an action space.
+Here is the declaration of `GridWorldEnv` and the implementation of `__init__`:
 ```python
-import gym
-from gym import error, spaces, utils
-from gym.utils import seeding
+class GridWorldEnv(gym.Env):
+    metadata = {"render_modes": ["human", "rgb_array"], "render_fps": 4}
 
-class FooEnv(gym.Env):
-    metadata = {'render.modes': ['human']}
+    def __init__(self, size=5):
+        self.size = size    # The size of the square grid
+        self.window_size = 512  # The size of the PyGame window
+        
+        # Observations are dictionaries with the agent's and the target's location.
+        # Each location is encoded as an element of {0, ..., `size`}^2, i.e. MultiDiscrete([size, size]).
+        self.observation_space = spaces.Dict(
```

Comment:
This isn't necessarily me requesting a change, just thinking out loud -- is a `Dict[MultiDiscrete]` space the best choice here? It uses two relatively niche features which might be annoying for simple RL implementations. 

On the one hand, it actually showcases how to use these niche features. On the other hand, here it might be better to focus on `Discrete` and `Box` which (from my perception) make up like 90% of the use cases

Reply:
If we want to use a more common space instead, it can be `Discrete(25*25)` or `Box(low=0, high=4, shape=(4,))` (with either some `int` as the dtype, or just `float32` to be practical)

Reply:
I agree that the observation space is not optimal and I wasn't completely comfortable with it either. I'm not a big fan of `Discrete(25*25)` because you'd lose all structure of the observations and computing them would be a nightmare. I think `Box` with `dtype=int` is a good idea. I'm mostly indifferent to whether it's a dictionary of boxes or one large `Box` (via concatenation). But I lean slightly towards making it a dictionary because it's more readable and shows off an additional feature. And it's always possible to add a `FlattenObservation` wrapper on top later (or even [this example wrapper from the docs](https://www.gymlibrary.ml/content/wrappers/#observationwrapper)).